### PR TITLE
leiden nmi ari

### PR DESCRIPTION
**Pull request by adamgayoso**
Adds leiden nmi ari scores to more closely match scib (they use louvain). This does a search of 10 resolutions of leiden clustering to pick the optimal NMI (as in scIB, but it uses 20 res params)

### CODE REVIEW

File: tests/test_basic.py [hunk 1]
```diff
@@ -47,9 +49,23 @@ def test_silhouette_batch():
...
     X, labels = dummy_x_labels()
-    nmi, ari = scib_metrics.nmi_ari_cluster_labels(X, labels)
+    nmi, ari = scib_metrics.nmi_ari_cluster_labels_kmeans(X, labels)
+    assert isinstance(nmi, float)
+    assert isinstance(ari, float)
+
+
+def test_nmi_ari_cluster_labels_leiden_parallel():
+    X, labels = dummy_x_labels(return_symmetric_positive=True)
+    nmi, ari = scib_metrics.nmi_ari_cluster_labels_leiden(X, labels, optimize_resolution=True, n_jobs=2)
```
Comment (justjhong) [inline tests/test_basic.py:61]: is it possible to check this against some leiden impl to make sure the clusters are the same? Is this reliable for diff random seeds?

Comment (justjhong) [reply tests/test_basic.py:61]: ah jk i see its leiden vs louvain

File: src/scib_metrics/_ari_nmi.py [hunk 2]
```diff
@@ -33,9 +46,72 @@ def nmi_ari_cluster_labels(X: np.ndarray, labels: np.ndarray) -> Tuple[float, fl
...
+        resolutions = np.array([2 * x / n for x in range(1, n + 1)])
+        try:
+            from joblib import Parallel, delayed
+
+            out = Parallel(n_jobs=n_jobs)(
+                delayed(nmi_ari_cluster_labels_leiden)(X, labels, False, r) for r in resolutions
+            )
+        except ImportError:
+            logger.info("Using for loop over resolutions. pip install joblib for parallelization.")
+            out = [nmi_ari_cluster_labels_leiden(X, labels, False, r) for r in resolutions]
```
Comment (justjhong) [inline src/scib_metrics/_ari_nmi.py:107]: this recursive pattern feels like a recipe for bugs. would be much cleaner to just split out the else case into a helper then call that from both cases (optimize or not optimize)

File: tests/test_basic.py (hunk 1 above)
Comment (adamgayoso) [reply tests/test_basic.py:61]: leiden is faster. And the igraph implementation is even faster than the one scanpy uses

File: src/scib_metrics/_ari_nmi.py (hunk 2 above)
Comment (adamgayoso) [reply src/scib_metrics/_ari_nmi.py:107]: Done.

**Review by justjhong — APPROVED** [review]

File: src/scib_metrics/_ari_nmi.py (hunk 2 above)
Comment (justjhong) [reply src/scib_metrics/_ari_nmi.py]: speedy

**Pull request merged**